From cea6a569afa544826c91fda7ef0f9941f50c9459 Mon Sep 17 00:00:00 2001
From: Rui Ueyama <ruiu@cs.stanford.edu>
Date: Wed, 12 Jan 2022 16:31:04 +0900
Subject: [PATCH] [ELF] Remove "GNU gold" from the --version string

That substring was introduced in de7ba9014f9d9d70b9173c4bfac6c3433186ba49
to appease Gentoo's dev-libs/jansson-2.13.1-r1 package. But I think I
shouldn't done that from the beginning to avoid the sitaution of the
"User-Agent" string of the web browser, which everybody claims they
are Mozilla, AppleWebKit, Chrome and Safari simultaneously.
---
 main.cc             | 4 ++--
 test/elf/version.sh | 6 +++---
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/main.cc b/main.cc
index 38a41cdf..8831f7f2 100644
--- a/main.cc
+++ b/main.cc
@@ -26,10 +26,10 @@ std::string_view errno_string() {
 
 #ifdef GIT_HASH
 const std::string mold_version =
-  "mold " MOLD_VERSION " (" GIT_HASH "; compatible with GNU ld and GNU gold)";
+  "mold " MOLD_VERSION " (" GIT_HASH "; compatible with GNU ld)";
 #else
 const std::string mold_version =
-  "mold " MOLD_VERSION " (compatible with GNU ld and GNU gold)";
+  "mold " MOLD_VERSION " (compatible with GNU ld)";
 #endif
 
 void cleanup() {
diff --git a/test/elf/version.sh b/test/elf/version.sh
index ae52b504..4ca576ef 100755
--- a/test/elf/version.sh
+++ b/test/elf/version.sh
@@ -10,10 +10,10 @@ mold="$(pwd)/mold"
 t=out/test/elf/$testname
 mkdir -p $t
 
-"$mold" -v | grep -q 'mold .*compatible with GNU ld and GNU gold'
-"$mold" --version | grep -q 'mold .*compatible with GNU ld and GNU gold'
+"$mold" -v | grep -q 'mold .*compatible with GNU ld'
+"$mold" --version | grep -q 'mold .*compatible with GNU ld'
 
-"$mold" -V | grep -q 'mold .*compatible with GNU ld and GNU gold'
+"$mold" -V | grep -q 'mold .*compatible with GNU ld'
 "$mold" -V | grep -q elf_x86_64
 "$mold" -V | grep -q elf_i386
 
